components/tidb_query/src/codec/mysql/duration.rs

```diff
@@ -140,6 +140,56 @@ mod parser {
         Ok((rest, hhmmss))
     }
 
+    fn hhmmss_datetime(input: &str, fsp: u8) -> IResult<&str, [u32; 4], ()> {
+        let (rest, digits) = digit1(input)?;
+        if digits.len() == 12 || digits.len() == 14 {
```

Comment:
I think we shouldn't assume the length of digital string and it's the duty of `DateTime::parse_datetime`.